crates/swc_css_parser/src/lexer/mod.rs

```diff
@@ -1007,19 +996,19 @@ where
         maybe_third: Option<char>,
     ) -> LexResult<bool> {
         // Look at the first code point:
-        let first = maybe_first.or(self.input.cur());
+        let first = maybe_first.or(self.cur());
```

Comment:
I think this will call `self.next()` unconditionally.

Reply:
@kdy1 Why? We should look here on currrent and the next if no options were provided, from spec:

> This section describes how to check if three code points would start an identifier. The algorithm described here can be called explicitly with three code points, or can be called with the input stream itself. In the latter case, the three code points in question **are the current input code point and the next two input code points**, in that order.

Reply:
I meant  `.or` should be changed to `.or_else` to avoid useless utf8 decoding for cases where `first` is `Some`.

But nevermind, I found that original source code was the same.  It can be handled with another PR.

Reply:
Ah, I see, let's fix it, I can fix it here or we can postpone it to the next PR